app/components/thumbnail_component.rb

```diff
@@ -3,26 +3,47 @@
 class ThumbnailComponent < ApplicationComponent
   attr_reader :resource, :featured
 
-  # @param [SolrDocument, WorkVersion, Collection] resource
+  # @resource could be a SolrDocument or SolrDocumentAdapterDecorator (containing a Collection or Work),
+  # Collection, Work, WorkVersion, CollectionDecorator, WorkDecorator, or WorkVersionDecorator
   def initialize(resource:, featured: false)
     @resource = resource
     @featured = featured
   end
 
-  def icon
-    icon_map.fetch(icon_key, 'bar_chart')
+  def thumbnail
+    if thumbnail_url.present?
+      "<img src=#{thumbnail_url}></img>"
+    else
+      "<i class='material-icons material-icons--thumbnail'>#{icon}</i>"
+    end
```

Comment:
IMO, I think it might be cleaner to put this logic and the HTML generation inside `thumbnail_component.html.erb`. Then you won't have to use `html_safe` in the template. We do conditional rendering in other existing components like `files_message_component` and `work-version_detail_component`.